### DISCUSSION TITLE
Nightly build of the Control Gallery?

### DISCUSSION THREAD

**Discussion by 0x90d** [AvaloniaUI/Avalonia#10035] [Q&A]
The control gallery from Microsoft store is limited and basically not updated. The standalone is abandoned. Whats left is the control gallery in this repo here. However you cannot simply launch it. It only works if you fully build everything, which is a heavy task and requires a lot to be installed.

Are there any nightly builds of the control gallery available to download?

**Comment by timunie** [comment]
@0x90d which ControlGallery are you talking about? Can you share a link?

**Reply by 0x90d** [reply]
I'm talking about several galleries. The [Microsoft one](https://apps.microsoft.com/store/detail/avalonia-control-gallery/9PGHN6QNJ3RP?hl=en-us&gl=us) appears to be [this one](https://github.com/AvaloniaUI/XamlControlsGallery) but unlike the screenshot suggests it's actually missing tons of controls. Last commit to the repo is over a year ago.

It only has these few pages:
![image](https://user-images.githubusercontent.com/46010672/213996862-a95f6ebb-fddc-40a1-a78b-15e77e929a2c.png)

Then there is/was the [standalone gallery](https://github.com/AvaloniaUI/ControlCatalogStandalone). It was probably the most active gallery control repo but it has been shut down. I assume that was done so the control gallery can be moved into this repo, see [here](https://github.com/AvaloniaUI/Avalonia/tree/master/samples/ControlCatalog) and here its getting updates and all the new controls etc. The problem is that there is no standalone ready-to-use ready-to-look-at release, such like an executable for Windows. That wouldn't be a big problem if it would be using nuget packages but it don't. It requires that all other projects in this repo can also be build. And that isn't as easy as it has tons of custom build scripts. For me it fails every time. Starting with VS complains that I dont have .NET Framework 4.6 (even trying to run the .NET Core version, VS won't build because another project has targeted .NET Framework 4.6), fixing that then the powershell build script fails and so on. But even if I could get it to work, I also have to install VS packages I don't use like UWP, mobile etc. So building ALL projects its definitely not something I want to do.

While you provide nightly nuget packages of Avalonia, you don't provide nightly versions of the control gallery. And I cannot get any artifacts from the build server you're using, that could have been another workaround.

**Reply by timunie** [reply]
Thanks for the links.

**Reply by grokys** [reply]
We should either update the one in the MS store, or remove it. I'm guessing that no-one in the core team has time to maintain it, so we may want to look to the community to do so?

@danwalmsley you appear to have created that project, what do you think?

**Reply by rabbitism** [reply]
I wish the Ms store one can be updated, this is also a good promotion.
Otherwise, at least provide a demo executable download together with each release, as material xaml project did.

**Reply by 0x90d** [reply]
> at least provide a demo executable download together with each release

That would be fine as well.

**Reply by ahopper** [reply]
building the sample ControlCatalog for each platform with each CI build and providing links to the artefacts would be useful for quick sanity checks on prs and master.

**Reply by maxkatz6** [reply]
Our PR builds are slow already, having control catalog built for all platforms will increase build time multiple times more.

**Reply by ahopper** [reply]
fair enough, maybe there is someone generous enough with their time and cpu resources out there to do this for the project :) It would be a useful feature.
